app/src/main/java/de/westnordost/streetcomplete/quests/collection_times/AddCollectionTimes.java

```diff
@@ -0,0 +1,56 @@
+package de.westnordost.streetcomplete.quests.collection_times;
+
+import android.os.Bundle;
+import android.support.annotation.NonNull;
+import android.text.TextUtils;
+import android.util.Log;
+
+import java.util.Map;
+
+import javax.inject.Inject;
+
+import de.westnordost.streetcomplete.R;
+import de.westnordost.streetcomplete.data.osm.Countries;
+import de.westnordost.streetcomplete.data.osm.SimpleOverpassQuestType;
+import de.westnordost.streetcomplete.data.osm.download.OverpassMapDataDao;
+import de.westnordost.streetcomplete.quests.AbstractQuestAnswerFragment;
+import de.westnordost.streetcomplete.data.osm.changes.StringMapChangesBuilder;
+
+public class AddCollectionTimes extends SimpleOverpassQuestType
+{
+	@Inject public AddCollectionTimes(OverpassMapDataDao overpassServer) { super(overpassServer); }
+
+	@Override protected String getTagFilters()
+	{
+		String[] amenities = { "post_box" };
+
+		return " nodes, ways, relations with ( amenity ~ "
+				+ TextUtils.join("|", amenities) + ")" +
+				" and !collection_times" + " and (access !~ private|no)"; // exclude ones without access to general public
+	}
+
+	@Override public AbstractQuestAnswerFragment createForm()
+	{
+		return new AddCollectionTimesForm();
+	}
+
+	@Override public void applyAnswerTo(Bundle answer, StringMapChangesBuilder changes)
+	{
+		String collectionTimes = answer.getString(AddCollectionTimesForm.COLLECTION_TIMES);
+		if(collectionTimes != null)
+		{
+			changes.add("collection_times", collectionTimes);
+		}
+	}
+
+	@NonNull @Override public Countries getEnabledForCountries()
+	{
+		return Countries.allExcept(new String[]{
+			""
```

Comment:
In #85 @westnordost said:

> OK this quest may make sense, BUT only in those countries where a timetable is actually (usually) visible on every post box. Otherwise, it creates a lot of false positives.
What I need here before is a list of countries in which is the case.

I think this is actually the harder part of the implementation... Before creating a new quest, we should consider in which countries it makes sense to ask. So far we only know that it makes sense in:

- Germany
- UK
- Denmark
- Austria
- Switzerland
- Sweden
- France
- Poland
- Netherlands
- USA?
- Canada?

Reply:
I can confirm USA. Each of the USPS (US Postal Service) post boxes I've been to in Washington state has had collection times labeled in front, and because USPS is national, it should be the same in every other state.